Bang.Tests/DomainUnitTests/ServerMessageProcessorTests.cs

```diff
@@ -0,0 +1,224 @@
+            Assert.True(responseJoinGameMessage.IsJoined);
+        }
+
+        [Fact]
+        public void Cannot_join_into_full_game()
+        {
+            var player = CreatePlayer();
+            var game = CreateGame(player);
+            for (int i = 0; i < 6; i++)
+            {
+                game.Players.Add(player);
+            }
+
+            var joinGameMessage = new JoinGameMessage();
+            joinGameMessage.PlayerId = player.Id;
+            joinGameMessage.GameId = game.Id;
+
+            var serverProcessor = new ServerMessageProcessor();
+            var response = serverProcessor.ProcessJoinGameMessage(joinGameMessage);
+            var responseJoinGameMessage = response.First() as JoinGameMessage;
+
+            Assert.False(responseJoinGameMessage.IsJoined);
+        }
+
+        [Fact]
+        public void Player_can_leave_game()
+        {
+            var player = CreatePlayer();
+            var game = CreateGame(player);
+
+            var player2 = CreatePlayer();
+            game.JoinPlayer(player2);
+
+            var leaveGameMessage = new LeaveGameMessage();
+            leaveGameMessage.GameId = game.Id;
+            leaveGameMessage.PlayerId = player2.Id;
+
+            var serverProcessor = new ServerMessageProcessor();
+            serverProcessor.ProcessLeaveGameMessage(leaveGameMessage);
+
+            Assert.DoesNotContain(player2, game.Players);
+        }
+
+        [Fact]
+        public void When_the_last_player_leaves_the_game_closes()
+        {
+            var player = CreatePlayer();
+            var game = CreateGame(player);
+            
+            var leaveGameMessage = new LeaveGameMessage();
+            leaveGameMessage.GameId = game.Id;
+            leaveGameMessage.PlayerId = player.Id;
+
+            var serverProcessor = new ServerMessageProcessor();
+            serverProcessor.ProcessLeaveGameMessage(leaveGameMessage);
+
+            Assert.DoesNotContain(game, Lobby.GetGames());
+        }
+
+        [Fact]
+        public void Ready_to_play_message_changes_player_status()
+        {
+            var player = CreatePlayer();
+            var game = CreateGame(player);
+
+            var readyToPlayMessage = new ReadyToPlayMessage();
+            readyToPlayMessage.GameId = game.Id;
+            readyToPlayMessage.PlayerId = player.Id;
+            readyToPlayMessage.IsReady = true;
+
+            var serverProcessor = new ServerMessageProcessor();
+            serverProcessor.ProcessReadyToPlayMessage(readyToPlayMessage);
+            Assert.True(game.Players.First(p => p.Id == player.Id).IsReadyToPlay);
+
+            readyToPlayMessage.IsReady = false;
+            serverProcessor.ProcessReadyToPlayMessage(readyToPlayMessage);
+            Assert.False(game.Players.First(p => p.Id == player.Id).IsReadyToPlay);
+        }
+
+        [Fact]
+        public void When_all_players_are_ready_initialize_game()
```

Comment:
Mutliple asserts in unit test are sign that test checks a lot of things. It makes sense to divide the test into two or three

Reply:
done